embassy-net/src/tcp.rs

```diff
@@ -79,6 +79,9 @@ impl<'a> TcpReader<'a> {
     ///
     /// Returns how many bytes were read, or an error. If no data is available, it waits
     /// until there is at least one byte available.
+    ///
+    /// A return value of Ok(0) means that the socket was closed and is longer able to
```

Comment:
"closed" in TCP is a bit ambiguous. It can be sending a FIN, receiving a FIN, or a RST. 

`Ok(0)` denotes EOF, which in TCP means the remote side has closed the "receive half" of the connection by sending us a FIN, and we've read all the data (ie there's no more data).

This is different than a close by RST (that you do with `.abort()`). In that case `read()` returns `Err(ConnectionReset)`. 

And it's also different than calling `.close()`. That closes the "transmit half" of the connection, ie we can't *send* more bytes, but the receive half stays open.

Can you expand the docs a bit to clarify that? thanks!

Reply:
thanks for the explanation, ill work it into the docs :+1: